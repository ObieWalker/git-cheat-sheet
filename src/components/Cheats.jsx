import React from 'react';
import { OverlayTrigger, Button, Tooltip } from 'react-bootstrap'

const Cheats = ({cheat, onCopy, deleteCheat, userId, categoryId }) => {
  const tooltip = (
    <Tooltip id="tooltip">
      <strong>Click to Copy</strong>.
    </Tooltip>
  );
  return (
      <div>
        {(cheat.userId === null || cheat.userId === userId) &&
        <div>
          <h5 className="white-text">{cheat.description}
          {cheat.userId &&
          <button 
            className="btn-sm btn-danger pull-right"
            onClick={() => deleteCheat(cheat._id, categoryId)} >X
            </button>}
          </h5>
          <OverlayTrigger placement="bottom" overlay={tooltip}>
            <h6
              bsStyle="default"
              className="white-text onCopy"          
              onClick={() => onCopy(cheat.command)}
            >$ {cheat.command}</h6>
          </OverlayTrigger>
          <br />
        </div>
        }
      </div>
    );
  }

export default Cheats;
