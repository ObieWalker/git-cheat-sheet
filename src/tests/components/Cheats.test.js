import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import Cheats from '../../components/Cheats';

let mounted;
let props;
let wrapper;
let userId = '1234'
const cheat = {
  userId : userId
}
const onCopy = jest.fn(() => Promise.resolve({}));
const deleteCheat = jest.fn(() => Promise.resolve({}));
const categoryId = ''

const getComponent = () => {
  if (!mounted) {
    props = {
      cheat,
      onCopy,
      userId,
      deleteCheat,
      categoryId
    };
    mounted = shallow(<Cheats {...props} />);
  }
  return mounted;
};

describe('cheats Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should click on delete cheat button', () => {
    wrapper = getComponent();
    wrapper
    .find('.btn-sm')
    .simulate('click');
  });
  it('should click onCopy button', () => {
    wrapper = getComponent();
    wrapper
    .find('.onCopy')
    .simulate('click');
  });

});

