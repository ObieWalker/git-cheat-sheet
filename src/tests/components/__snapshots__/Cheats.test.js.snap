// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cheats Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cheats
    categoryId=""
    cheat={
      Object {
        "userId": "1234",
      }
    }
    deleteCheat={[MockFunction]}
    onCopy={[MockFunction]}
    userId="1234"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h5
          className="white-text"
        >
          <button
            className="btn-sm btn-danger pull-right"
            onClick={[Function]}
          >
            X
          </button>
        </h5>
        <OverlayTrigger
          defaultOverlayShown={false}
          overlay={
            <Tooltip
              bsClass="tooltip"
              id="tooltip"
              placement="right"
            >
              <strong>
                Click to Copy
              </strong>
              .
            </Tooltip>
          }
          placement="bottom"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <h6
            bsStyle="default"
            className="white-text onCopy"
            onClick={[Function]}
          >
            $ 
          </h6>
        </OverlayTrigger>
        <br />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h5
            className="white-text"
          >
            <button
              className="btn-sm btn-danger pull-right"
              onClick={[Function]}
            >
              X
            </button>
          </h5>,
          <OverlayTrigger
            defaultOverlayShown={false}
            overlay={
              <Tooltip
                bsClass="tooltip"
                id="tooltip"
                placement="right"
              >
                <strong>
                  Click to Copy
                </strong>
                .
              </Tooltip>
            }
            placement="bottom"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <h6
              bsStyle="default"
              className="white-text onCopy"
              onClick={[Function]}
            >
              $ 
            </h6>
          </OverlayTrigger>,
          <br />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <button
                className="btn-sm btn-danger pull-right"
                onClick={[Function]}
              >
                X
              </button>,
            ],
            "className": "white-text",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "X",
                "className": "btn-sm btn-danger pull-right",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "X",
              "type": "button",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h6
              bsStyle="default"
              className="white-text onCopy"
              onClick={[Function]}
            >
              $ 
            </h6>,
            "defaultOverlayShown": false,
            "overlay": <Tooltip
              bsClass="tooltip"
              id="tooltip"
              placement="right"
            >
              <strong>
                Click to Copy
              </strong>
              .
            </Tooltip>,
            "placement": "bottom",
            "trigger": Array [
              "hover",
              "focus",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "bsStyle": "default",
              "children": Array [
                "$ ",
                undefined,
              ],
              "className": "white-text onCopy",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "$ ",
              undefined,
            ],
            "type": "h6",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h5
            className="white-text"
          >
            <button
              className="btn-sm btn-danger pull-right"
              onClick={[Function]}
            >
              X
            </button>
          </h5>
          <OverlayTrigger
            defaultOverlayShown={false}
            overlay={
              <Tooltip
                bsClass="tooltip"
                id="tooltip"
                placement="right"
              >
                <strong>
                  Click to Copy
                </strong>
                .
              </Tooltip>
            }
            placement="bottom"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <h6
              bsStyle="default"
              className="white-text onCopy"
              onClick={[Function]}
            >
              $ 
            </h6>
          </OverlayTrigger>
          <br />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="white-text"
            >
              <button
                className="btn-sm btn-danger pull-right"
                onClick={[Function]}
              >
                X
              </button>
            </h5>,
            <OverlayTrigger
              defaultOverlayShown={false}
              overlay={
                <Tooltip
                  bsClass="tooltip"
                  id="tooltip"
                  placement="right"
                >
                  <strong>
                    Click to Copy
                  </strong>
                  .
                </Tooltip>
              }
              placement="bottom"
              trigger={
                Array [
                  "hover",
                  "focus",
                ]
              }
            >
              <h6
                bsStyle="default"
                className="white-text onCopy"
                onClick={[Function]}
              >
                $ 
              </h6>
            </OverlayTrigger>,
            <br />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <button
                  className="btn-sm btn-danger pull-right"
                  onClick={[Function]}
                >
                  X
                </button>,
              ],
              "className": "white-text",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "btn-sm btn-danger pull-right",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
            ],
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h6
                bsStyle="default"
                className="white-text onCopy"
                onClick={[Function]}
              >
                $ 
              </h6>,
              "defaultOverlayShown": false,
              "overlay": <Tooltip
                bsClass="tooltip"
                id="tooltip"
                placement="right"
              >
                <strong>
                  Click to Copy
                </strong>
                .
              </Tooltip>,
              "placement": "bottom",
              "trigger": Array [
                "hover",
                "focus",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "bsStyle": "default",
                "children": Array [
                  "$ ",
                  undefined,
                ],
                "className": "white-text onCopy",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "$ ",
                undefined,
              ],
              "type": "h6",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
